### PR TITLE
Update versioning governance document

### LINKED ISSUE #21073: Versioning Policy Update
### Outline & Motivation

Firstly, thanks to Lightning team for your continued hard work and invaluable contributions to the open-source ecosystem! 

As a follow-up to [this issue](https://github.com/Lightning-AI/pytorch-lightning/issues/20670),  I’ve noticed that recent Lightning minor releases are no longer synchronized with PyTorch’s minor releases as they have been in the past:

| PyTorch Version | Release Date   | Release Artifact                                                                | Lightning Version | Release Date      | Release Artifact                                                                                     | Lag (weeks: Lightning - PyTorch) |
|:---------------:|:--------------|:--------------------------------------------------------------------------------|:-----------------:|:------------------|:-----------------------------------------------------------------------------------------------------|:---------------------------------:|
| 2.0             | Mar 15, 2023  | [PyTorch 2.0.0](https://github.com/pytorch/pytorch/releases/tag/v2.0.0)         | 2.0.0             | Mar 15, 2023      | [Lightning 2.0.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.0.0)              | 0                                |
| 2.1             | Oct 4, 2023   | [PyTorch 2.1.0](https://github.com/pytorch/pytorch/releases/tag/v2.1.0)         | 2.1.0             | Oct 12, 2023      | [Lightning 2.1.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.1.0)              | 1                                |
| 2.2             | Jan 30, 2024  | [PyTorch 2.2.0](https://github.com/pytorch/pytorch/releases/tag/v2.2.0)         | 2.2.0             | Feb 7, 2024       | [Lightning 2.2.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.2.0)              | 1                                |
| 2.3             | Apr 24, 2024  | [PyTorch 2.3.0](https://github.com/pytorch/pytorch/releases/tag/v2.3.0)         | 2.3.0             | Jun 13, 2024      | [Lightning 2.3.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.3.0)              | 7 (50 days)                      |
| 2.4             | Jul 24, 2024  | [PyTorch 2.4.0](https://github.com/pytorch/pytorch/releases/tag/v2.4.0)         | 2.4.0             | Aug 7, 2024       | [Lightning 2.4.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.4.0)              | 2 (14 days)                      |
| 2.5             | Oct 17, 2024  | [PyTorch 2.5.0](https://github.com/pytorch/pytorch/releases/tag/v2.5.0)         | 2.5.0             | Dec 20, 2024      | [Lightning 2.5.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.0)              | 9 (64 days)                      |
| 2.6             | Jan 29, 2025  | [PyTorch 2.6.0](https://github.com/pytorch/pytorch/releases/tag/v2.6.0)         | —                 | —                 | —                                                                                                    | 28 (197 days)                    |
| 2.7             | Apr 23, 2025  | [PyTorch 2.7.0](https://github.com/pytorch/pytorch/releases/tag/v2.7.0)         | —                 | —                 | —                                                                                                    | 16 (113 days)                    |
| 2.8             | Aug 6, 2025   | [PyTorch 2.8.0](https://github.com/pytorch/pytorch/releases/tag/v2.8.0)         | —                 | —                 | —                                                                                                    | 1 (8 days)                       |

### Pitch

I think both users and maintainers of downstream packages depending on Lightning (I'm the primary maintainer of a downstream Lightning package [Fine-Tuning Scheduler](https://github.com/speediedan/finetuning-scheduler)) would benefit if this apparent versioning policy/cadence change were made explicit in the [versioning documentation](https://github.com/Lightning-AI/pytorch-lightning/blob/a9deadaf418ae85201a5e4b91642a50c82bfd328/docs/source-pytorch/versioning.rst). 

Can we update the versioning policy to clarify the current minor release policy/cadence for Lightning? E.g., _'minor releases are typically driven by key upstream PyTorch deprecations etc...'_ This would help the community better plan dependencies and upgrades. 

### Additional context

As a concrete example, Fine-Tuning Scheduler has a number of deprecations and enhancements aligned with various PyTorch minor versions that have been awaiting a Lightning minor release. If the release cadence of the last year is expected to be the new approach moving forward, it would be good to know so FTS can proceed with changing its release approach from being Lightning release synchronized to a PyTorch release alignment instead.
Thanks again!

@Borda @lantiga @iwr-redmond 

cc @lantiga @borda @justusschock

### PR DESCRIPTION
**Pull request by speediedan**
## What does this PR do?

This versioning governance document update makes the following clarifications and minor updates:
1. Clarifies the existing Lightning API evolution policy
2. Makes the Compatiblity Matrix discussion more explicit with respect to expected PyTorch support ranges (while retaining flexibility)
3. Bumps current ``2.5.x`` support range to include PyTorch ``2.8.0``

Updates to the versioning governance revolve around the discussion in [#21073](https://github.com/Lightning-AI/pytorch-lightning/issues/21073) and should resolve that issue.

Fixes #21073 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [X] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21107.org.readthedocs.build/en/21107/

### CODE REVIEW

File: docs/source-pytorch/versioning.rst [hunk 1]
```diff
@@ -82,7 +79,7 @@ The table below indicates the coverage of tested versions in our CI. Versions ou
    * - 2.5
      - 2.5
      - 2.5
-     - ≥2.1, ≤2.7
+     - ≥2.1, ≤2.8
```
Comment (Borda) [inline docs/source-pytorch/versioning.rst]:
```suggestion
     - ≥2.1, (last tested 2.8)
```

**Review by SkafteNicki — APPROVED** [review]
